### PR TITLE
useful standalone utility scripts

### PR DESCRIPTION
**Pull request by kamfonik**
Standalone scripts for getting information from OpenStack that isn't easily accessed through the CLI.

* `list_roles.py` 
mode 1: shows what projects a user belongs to, and the user's roles there
mode 2: shows what users belong to a project, and their roles
Why we need it:
    * OpenStack CLI `user role list` shows roles in only one project - either one you specify or the one you are authenticated to.  No way to discover which projects user X belongs to.
    * CLI has no simple way to list project members; you can see it in the GUI but there is much clicking because the project filter is broken in Horizon

* `search_ips.py`
Finds information about an external IP with just the IP address.  
Why we need it:
    * Sometimes we get a warning that some IP is generating suspicious traffic.  Finding out who owns the IP and what it's attached to via OpenStack CLI involes numerous steps.
    * Also sometimes users use the public IP to identify their VM to us when they ask for help.  We can go back and ask them for more info but this can be easier.

Both scripts are standalone and not related to user automation.  But they use the same settings.ini file as the rest of the scripts in here.

### CODE REVIEW

**Comment by kamfonik** [comment]
@xuhang57 this is low priority so if you are busy with summit stuff do that.  Maybe review if you need a break.

**Comment by xuhang57** [comment]
@kamfonik  Thanks. I will review this over the weekend. Have to focus on the sahara in GIJI today.

File: utility_scripts/list_roles.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+admin_user = config.get('auth', 'admin_user')
+admin_pwd = config.get('auth', 'admin_pwd')
+admin_project = config.get('auth', 'admin_project')
+auth_url = config.get('auth', 'auth_url')
+nova_version = config.get('nova', 'version')
+
+auth = v3.Password(auth_url=auth_url,
+                   username=admin_user,
+                   user_domain_id='default',
```
Comment (xuhang57) [inline utility_scripts/list_roles.py]: ·default` is good for now. But you might want to have a global variable that is called `domain`. So later on, if we support different domains, you could just change that global variable. (even though I have no idea whether we are going to support different domains at all or not?)

Comment (kamfonik) [inline utility_scripts/list_roles.py]: good point, and we want other projects to be able to use this code if they want, maybe their domain is not called 'default'.

**Comment by kamfonik** [comment]
@jeremyfreudberg yes, I agree, we need to reduce duplication across this whole repo.  I think CLI and config loading are the major points there but I suspect we can also create one base parse_rows()

**Pull request merged**